frontend/src/services/baseService.ts

```diff
@@ -1,12 +1,15 @@
 import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
 import { GetRegisteredModelsResponse, ModelInfo } from '../api/index.ts';
 
+
+const isDevServer = window.location.host.endsWith(":5174");
+
 export const baseServiceAPI = createApi({
     reducerPath: 'baseServiceAPI',
     tagTypes: [
       "models",
     ],
-    baseQuery: fetchBaseQuery({ baseUrl: 'http://127.0.0.1:8000/v1/' }),
+    baseQuery: fetchBaseQuery({ baseUrl: isDevServer ? "http://0.0.0.0:8000/v1" : '/v1/' }),
```

Comment:
Dev-server is a special case, in general the frontend should be served from the same hostname as the API, so we omit hostname.